src/storage/store/engine.rs

```diff
@@ -963,32 +964,67 @@ impl ShardEngine {
         Ok(())
     }
 
+    fn get_username_proof(
+        &self,
+        name: String,
+        txn: &mut RocksDbTransactionBatch,
+    ) -> Result<Option<UserNameProof>, MessageValidationError> {
+        // TODO(aditi): The fnames proofs should live in the username proof store.
+        if name.ends_with(".eth") {
+            let proof_message = UsernameProofStore::get_username_proof(
+                &self.stores.username_proof_store,
+                &name.encode_to_vec(),
+                UserNameType::UsernameTypeEnsL1 as u8,
+            )
+            .map_err(|e| MessageValidationError::StoreError {
+                inner: e,
+                hash: vec![],
+            })?;
+            match proof_message {
+                Some(message) => match message.data {
+                    None => Ok(None),
+                    Some(message_data) => match message_data.body {
+                        Some(body) => match body {
+                            proto::message_data::Body::UsernameProofBody(user_name_proof) => {
+                                Ok(Some(user_name_proof))
+                            }
+                            _ => Ok(None),
+                        },
+                        None => Ok(None),
+                    },
+                },
+                None => Ok(None),
+            }
+        } else {
+            UserDataStore::get_username_proof(&self.stores.user_data_store, txn, name.as_bytes())
```

Comment:
Let us add a check in the userdata store to panic if the name ends with .eth, and a todo to move it to the username proof store.